Analyzed the tests for the class TaskExecutor

francislaus opened the pull request:
This answered the following questions:

* What are the tests?
* Do the tests cover the requirements?
* Does your refactoring introduce anything that requires more tests?

This should close #11

francislaus committed:
Added first part of test documentation for #11

Added the introduction part of the test documentation to the report and
started to enlist and describe all functions

francislaus committed:
Added documentation of the tests for #11

Completed the section in the report file for the tests. All test
functions are enlisted and explained seperately. This commit answers the
question 'What are the tests?'

francislaus committed:
Added more documentation of the tests for #11

Added an paragraph that focuses on the question 'Does your refactoring introduce anything that requires more tests?'

francislaus committed:
Added template for requirements for #11

Added a template to answer the question "Do the tests cover the requirements?"

francislaus committed:
Updated template for #11

Updated the table template for the requirements

francislaus committed:
Changed the template for #11

Changed the template for the requirements and filled in the necessary
information.

francislaus committed:
Added additional description for #11

Added an descriptive sentence for the requirements table so that it
applies for all cases

apeinot reviewed (changes requested):
In the last paragraph, the two first sentences are contradictory. To be fix

Samuelhertzberg reviewed (changes requested):
"Therefore, we do not need to add extra test cases." To be removed
"6/7" => "six out of seven"

francislaus committed:
Improved text and table for #11

Improved the description text and the corresponding table for the
requirements met section

francislaus commented:
Made changes accordingly to the comments. I was not quite sure about the comments by @apeinot, I hope I did what you requested

Samuelhertzberg committed:
Added description of new test case.

Analyzed the new test case, now it fulfills criteria.

Jonagoo commented on report.md:
```diff
@@ -55,8 +55,10 @@ The following table will cover all requirements as they are described in previou
+## New test cases relating to refactored code
 
-
+One new test was added. The input data to the getTaskExecutor() function is set to entirely `null`. This should result in a nullPointerException, which the new test catch. If no nullPointerException is thrown, something is very wrong and the new test fails.
```
catches*

francislaus replied:
*NullPointerException

emilelm96 reviewed (commented):
NullPointerException should be spelled with a big N.

Samuelhertzberg committed:
Fixed stupid typos

NullPointerException is a class, test case is he/she/it and @Jonagoo is the *grammar man*.
